core/frontend/src/main/scala/io/udash/core/Definitions.scala

```diff
@@ -31,74 +31,90 @@ trait Presenter[S <: State] {
+  * The ViewFactory has to prepare model, [[io.udash.core.View]], [[io.udash.core.Presenter]] and link them together.
   * @tparam S State for which this pair is defined.
   */
-trait ViewPresenter[S <: State] {
+trait ViewFactory[S <: State[_]] {
   /**
     * Factory method which should return ready to used instance of [[io.udash.core.Presenter]] and [[io.udash.core.View]].
     * @return pair of presenter and view for state S
     */
   def create(): (View, Presenter[S])
 }
 
+/** [[ViewFactory]] creating [[ContainerView]]. */
+trait ContainerViewFactory[S <: ContainerState[_]] extends ViewFactory[S] {
+  override def create(): (ContainerView, Presenter[S])
+}
+
+/** [[ViewFactory]] creating [[FinalView]]. */
+trait FinalViewFactory[S <: FinalState[_]] extends ViewFactory[S]{
+  override def create(): (FinalView, Presenter[S])
+}
+
 /**
-  * Abstract view which should be used in order to implement View for [[io.udash.core.ViewPresenter]].
+  * Abstract view which should be used in order to implement View for [[io.udash.core.ViewFactory]].
   * The View implementation usually gets the model and the [[io.udash.core.Presenter]] as constructor arguments.
   */
 trait View {
   /**
-    * Will be invoked by [[io.udash.routing.RoutingEngine]] in order to render the child view inside
-    * the parent view. <br/><br/>
-    *
-    * <b>This method can receive null as "view" argument, then previous child view should be removed.</b>
-    *
-    * @param view view which origins from child
-    */
-  def renderChild(view: View): Unit
-
-  /**
     * Implementation of this method should return DOM representation of view.
     * @return DOM representation of view
     */
   def getTemplate: Modifier[Element]
 
-  def apply() = getTemplate
+  @deprecated("Use `getTemplate` instead.", "0.6.0")
+  def apply(): Modifier[Element] = getTemplate
 }
 
-/** A [[io.udash.core.View]] which does not have any child view. */
-trait FinalView extends View {
-  override def renderChild(view: View): Unit =
-    if (view != null) throw View.UnexpectedChildView
-}
+/** A [[io.udash.core.View]] which can render child view. */
+trait ContainerView extends View {
+  import scalatags.JsDom.all.div
 
-object View {
-  case object UnexpectedChildView extends RuntimeException("This view is not expected to render child view. Did you forget to implement renderChild method?")
+  /** Default implementation renders child views inside this element. */
+  protected val childViewContainer: Element = div().render
+
+  /**
+    * Will be invoked by [[io.udash.routing.RoutingEngine]] in order to render the child view inside
+    * the parent view. <br/><br/>
+    *
+    * <b>This method can receive `None` as "view" argument, then previous child view should be removed.</b>
+    *
+    * @param view view which origins from child
+    */
+  def renderChild(view: Option[View]): Unit = {
```

Comment:
Do we want the users to override this? If not, let's make it final and update the doc to state the behaviour, not wish for it.